### PR TITLE
Use correct deleter for UBspline_3d_d

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

This is part of #3401

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix

### Does this introduce a breaking change?

- No

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes/No. This PR is up to date with current the current state of 'develop'
- Yes/No. Code added or changed in the PR has been clang-formatted
- Yes/No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes/No. Documentation has been added (if appropriate)

### CODE REVIEW

File: tests/models/CMakeLists.txt [hunk 1]
```diff
@@ -34,7 +34,7 @@ if(NOT QMC_CUDA)
   simple_run_and_check(short-sho-grad_lap "${qmcpack_SOURCE_DIR}/tests/models/sho" grad_lap.xml 1 1 check_grad_lap.py)
 
   qmc_run_and_check(
-    short-grid_ext_pot-vmc
```
Comment (ye-luo) [inline tests/models/CMakeLists.txt:37]: this test runs 256 blocks. It is not a deterministic test

**Review by ye-luo — CHANGES_REQUESTED** [review]

**Comment by ye-luo** [comment]
@quantumsteve When you rebase and force push. github doesn't send notification. So please add a comment when you update this PR.

File: tests/models/CMakeLists.txt (hunk 1 above)
Comment (quantumsteve) [reply tests/models/CMakeLists.txt:37]: How might it be converted to a deterministic test? Other ideas to create a unit or deterministic test that covers this memory error?

Comment (ye-luo) [reply tests/models/CMakeLists.txt:37]: Open an issue and ping @Hyeondeok-Shin, he can help you out

Comment (markdewing) [reply tests/models/CMakeLists.txt:37]: I'm not sure a deterministic test /unit test to catch this memory error is worth the effort in this case.   This is a memory leak (I assume?) from a "put" method, which is only going to be called once per run.

Comment (prckent) [reply tests/models/CMakeLists.txt]: Extra motivation: ongoing test coverage in CI.

**Comment by quantumsteve** [comment]
I removed testing changes and created issue #3426.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
@quantumsteve do we still need this PR in draft?

**Comment by quantumsteve** [comment]
> @quantumsteve do we still need this PR in draft?

I removed the in draft label.

**Pull request merged**